docs/electric-api.yaml

```diff
@@ -159,6 +159,23 @@ paths:
 
                 Must be provided as the `shape_id` parameter when making
                 subsequent requests where `offset` is not `-1`.
+            x-electric-schema:
+              schema:
+                type: object
+                example: { "id": { "type": "int4", "dimensions": 0 }, "title": { "type": "text", "dimensions": 0 }, "status": { "type": "text", "dimensions": 0, "max_length": 8 } }
+                description: |-
+                  Object mapping column names to the corresponding schema object.
+                  The schema object contains the type of the column, the number of dimensions, and possibly additional properties.
+                  Non-array types have a dimension of `0`, while array types have a dimension of 1 or more.
+                  For instance, an array of booleans would have a type of `bool` and a dimension of `1`.
+                  Some types provide additional properties,
+                  e.g.: `VARCHAR(8)` has an additional `“max_length": 8` property,
+                        `BPCHAR(9)` has an additional `"length": 9` property,
+                        `TIME(3)` has an additional `"precision": 3` property,
+                        `NUMERIC(8,5)` has additional `"precision": 8` and `"scale": 5` properties,
+                        `INTERVAL(4)` has an additional `"precision": 4` property,
+                        `INTERVAL MINUTE TO SECOND` has an additional `"fields": "MINUTE TO SECOND"` property,
+                        `BIT(5)` has an additional `"length": 5` property.
```

Comment:
Does this generalise to Postgres types from extensions, are you able to extract the attribute names for them?

So for the pgvector `vector(256)` can you determine automatically that the parameter is also dimensions?

If this is hard coded (skimming the code it looks like it is) and can't be generalised, a tuple of arguments may be better?

Reply:
So PG typically stores information about type modifiers in `pg_attribute.atttypmod` so that's where we get this information from. It would be interesting to see if for types from extensions, e.g. `vector(256)`, if that information is present and will be correctly fetched.

Reply:
It'd be useful to test a vector value, as well as enums and domain types. Not necessarily in this PR, though.

Reply:
<img width="849" alt="image" src="https://github.com/user-attachments/assets/02eeb703-05fd-4441-8804-1cd5764021de">

I see @kevin-dp mentioned not sending `formatted_type` above https://github.com/electric-sql/electric-next/pull/168#issuecomment-2247035793, but including it along with the raw values would make generating a schema very easy for the Postgres/PGlite client use

Reply:
Could we possibly also include which columns make up the PK?

Reply:
> It'd be useful to test a vector value, as well as enums and domain types. Not necessarily in this PR, though.

We have unit tests for enums, domain types, and complex types.
Not for extensions. As discussed privately with @samwillis it would make sense to also add the formatted type in the header as this would cover the case of extensions where the client can then parse the formatted type themselves.